src/utils/orgUtils.ts

```diff
@@ -65,6 +72,98 @@ export class OrgUtils {
         }
     }
 
+    public static async getCompactLayoutsForSObject(
+        sObjectName: string
+    ): Promise<Object> {
+        const org = await Org.create();
+        const conn = org.getConnection();
+
+        const result = await conn.request<Object>(
+            `/services/data/v59.0/sobjects/${sObjectName}/describe/compactLayouts`
+        );
+
+        return Promise.resolve(result);
+    }
+
+    public static async getCompactLayoutForSObject(
+        sObjectName: string,
+        recordTypeId: string
+    ): Promise<Object> {
+        const org = await Org.create();
+        const conn = org.getConnection();
+
+        const result = await conn.request<Object>(
+            `/services/data/v59.0/sobjects/${sObjectName}/describe/compactLayouts/${recordTypeId}`
+        );
+
+        return Promise.resolve(result);
+    }
+
+    public static async getCompactLayoutFieldsForSObject(
+        sObjectName: string
+    ): Promise<CompactLayoutField[]> {
+        try {
+            const fields: CompactLayoutField[] = [];
+
+            // Get all the compact layouts associated to this sObject first
+            let result = await this.getCompactLayoutsForSObject(sObjectName);
+
+            if (result) {
+                // sObject can have multiple compact layouts associated with it. Get the default.
+                const defaultCompactLayoutId =
+                    result['defaultCompactLayoutId' as keyof Object];
+
+                // Mapping tab
+                const recordTypeCompactLayoutMappings =
+                    result['recordTypeCompactLayoutMappings' as keyof Object];
+
+                // ID of compact layout need to be normalized
+                const recordTypeCompactLayoutMapping = (
+                    recordTypeCompactLayoutMappings as unknown as Array<Object>
```

Comment:
What does the intermediary coercion to `unknown` do here? Is that required?

Reply:
Similar to above without a hard cast Typescript would complain.
<img width="915" alt="image" src="https://github.com/salesforce/salesforcedx-vscode-mobile/assets/42476999/9fd36bef-d458-48db-9436-3529f969ae12">

Reply:
Oh I get the issue now. In this case, I'm going to recommend my previous pie-in-the-sky type coercion be done now. The code gets hard to read all over the place ,because we're basically hedging against defining the types we expect to work with in the response, in favor of "cast to any" statements all over the place, which in turn is punting on type safety.

Let's just create the types instead. From the code you've got, I can glean the following type definitions that represent the data we expect to get in these results:

```typescript
export type SObjectCompactLayoutMapping = {
    compactLayoutId: string;
    compactLayoutName: string;
    recordTypeId: string;
};

export type SObjectCompactLayouts = {
    defaultCompactLayoutId?: string;
    recordTypeCompactLayoutMappings: SObjectCompactLayoutMapping[];
    fieldItems: CompactLayoutField[];
};
```

I mean, we can tweak these, change type names, etc. But if we tack to an approach of returning `Promise<SObjectCompactLayouts>` instead of `Object`, we can access all of the expected properties as first-class citizens rather than casts to generic types.